### PR TITLE
chore!: breaking changes for v7.0

### LINKED ISSUE #925: Change the AppName from "CP Editor" to "cpeditor"
. The setting files will be placed in "xxx/xxx/xxx/cpeditor" instead of "xxx/xxx/xxx/CP Editor", and we'll move the setting files. This will also cause some other changes, for example, the class in xprop will change.

### PR DESCRIPTION
**Pull request by ouuan**
## Description

#769 except for YAML settings. Also, drop legacy setting migration from v6.7 and earlier.

## Related Issues / Pull Requests

takes over #934, closes #925 and #926.

## How Has This Been Tested?

On Arch Linux.

## Checklist

- [x] If needed, I have opened a pull request or an issue to update the [documentation](https://github.com/cpeditor/cpeditor.github.io).
- [x] If these changes are notable, they are documented in [CHANGELOG.md](https://github.com/cpeditor/cpeditor/blob/master/CHANGELOG.md).

### CODE REVIEW

**Comment by ouuan** [comment]
These changes were planned to be merged into v7.0 because new releases before it like v6.12 were expected. Now they can just be merged into master.

File: src/Settings/SettingsUpdater.cpp
Comment (coder3101) [inline src/Settings/SettingsUpdater.cpp:1]: What is the reason for this change?

Comment (ouuan) [reply src/Settings/SettingsUpdater.cpp:1]: They are pretty old. I think there is no need to always keep them.

**Review by coder3101 — APPROVED** [review]
